### PR TITLE
add remove time view

### PR DESCRIPTION
**Pull request by Shahbajkhan2023**
add the view for remove the time that is added by user.

### CODE REVIEW

File: vendor/views.py [hunk 1]
```diff
@@ -241,4 +242,10 @@ def error_response(self, form):
         return {
             'status': 'failed',
             'message': f"{form.cleaned_data['from_hour']}-{form.cleaned_data['to_hour']} already exists for this day!"
-        }
\ No newline at end of file
+        }
+    
+class RemoveOpeningHoursView(AjaxAuthenticationMixin, View):
+    def get(self, request, pk=None):
```
Comment (sraikwar94) [inline vendor/views.py:248]: why did you passed none for pk default value?

Comment (Shahbajkhan2023) [reply vendor/views.py:248]: The first thing is that using pk = None allows the method to run even if the primary key (pk) is not found. If pk = None, it helps to easily handle errors, allowing you to check whether a primary key has been provided or not.

Comment (sraikwar94) [reply vendor/views.py:248]: @Shahbajkhan2023 do you think these check not already available in django?

Comment (Shahbajkhan2023) [reply vendor/views.py:248]: Here we make a Url like this                                                                                                        path('opening-hours/remove/<int:pk>/', views.RemoveOpeningHoursView.as_view(), name='remove_opening_hours'),

Comment (sraikwar94) [reply vendor/views.py:248]: Buddy, do you understand why did you use pk=None because I couldn't find any use case her?

Comment (Shahbajkhan2023) [reply vendor/views.py:248]: for error handling but we don't need here .

Comment (sraikwar94) [reply vendor/views.py:248]: Okay, you can fix it or merge it with vulnerability :smiling_face_with_tear:

Comment (Shahbajkhan2023) [reply vendor/views.py:248]: okay

Comment (Shahbajkhan2023) [reply vendor/views.py:248]: class RemoveOpeningHoursView(AjaxAuthenticationMixin, View):
    def get(self, request, pk):
        hour = get_object_or_404(OpeningHour, pk=pk)
        hour.delete()
        return JsonResponse({'status': 'success', 'id': pk})                                              Sir, is this correct?

Comment (sraikwar94) [reply vendor/views.py:248]: yes please proceed

**Pull request merged**